test/TetrominoManager.ts

```diff
@@ -142,5 +145,57 @@ describe("TetrominoManager", () => {
         );
       });
     });
+
+    describe("Test getting wall kick offsets", () => {
+      for (let type = TetrominoType.Blank; type <
+        TetrominoType.NumTetrominoTypes; type += 1) {
+        for (let rotation = TetrominoRotation.O;
+          rotation < TetrominoRotation.NumTetrominoRotations;
+          rotation += 1) {
+          const offsets = TetrominoManager.getTetrominoWallKickOffsets(type,
+            rotation);
+          switch (type) {
+            case TetrominoType.Square:
+            /* Fallthrough */
+            case TetrominoType.I:
+            /* Fallthrough */
+            case TetrominoType.J:
+            /* Fallthrough */
+            case TetrominoType.L:
+            /* Fallthrough */
+            case TetrominoType.Z:
+            /* Fallthrough */
+            case TetrominoType.S:
+              assert.deepStrictEqual(offsets, WALL_KICK_COR_OFFSETS[rotation],
+                `Wall kick offset of type: ${type}}, rotation: ${rotation}
+                not correct`);
+              break;
+            case TetrominoType.T:
+              if (rotation === TetrominoRotation.O) {
+                const refOffsets = JSON.parse(JSON.stringify
+                  (WALL_KICK_COR_OFFSETS[rotation]));
+                refOffsets.splice(WALL_KICK_IMPOSSIBLE_CASE_T_O_INDEX, 1);
+                assert.deepStrictEqual(offsets, refOffsets,
+                  `Wall kick offset of type: ${type}}, rotation: ${rotation}
+                  not correct`);
+              } else if (rotation === TetrominoRotation.Z) {
+                const refOffsets = JSON.parse(JSON.stringify
+                  (WALL_KICK_COR_OFFSETS[rotation]));
+                refOffsets.splice(WALL_KICK_IMPOSSIBLE_CASE_T_Z_INDEX, 1);
+                assert.deepStrictEqual(offsets, refOffsets,
+                  `Wall kick offset of type: ${type}}, rotation: ${rotation}
+                  not correct`);
+              }
+              break;
+            case TetrominoType.Ghost:
+            case TetrominoType.Blank:
+            default:
+              assert.isEmpty(offsets, `Expecting offsets of type ${type} to be
+                empty`);
+              break;
```

Comment:
I think we can omit the last `break` in the default case. In the end, I don't mind either way so if you prefer to have it, all cool!

Reply:
I'd prefer keeping it just for code completeness. It doesn't really hurt to keep it as well so 😆